When a folder is set for auto upload with "upload existing files" unchecked, background worker is killed and no auto uploads occurs

### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Set a folder to auto upload, with "Also upload existing files" unchecked
2. Observe that auto upload doesn't work, with this entry in logs : `File-sync kill worker since syncedFolder (<id>) is not enabled!`

### Expected behaviour

Auto upload should work

### Actual behaviour

Auto upload worker is killed

### Android version

12

### Device brand and model

Oneplus 7t

### Stock or custom OS?

Stock

### Nextcloud android app version

20240704 (f-droid)

### Nextcloud server version

29.0.3

### Using a reverse proxy?

Yes

### Android logs

_No response_

### Server error logs

_No response_

### Additional information

I think (based on a quick look to the code, I might be wrong) the issue comes from the check done [here](https://github.com/nextcloud/android/blob/master/app/src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt#L162-L165):

```kotlin
        if (!setSyncedFolder(syncedFolderID)) {
            Log_OC.d(TAG, "File-sync kill worker since syncedFolder ($syncedFolderID) is not enabled!")
            return true
        }
```

In [`setSyncedFolder`](https://github.com/nextcloud/android/blob/master/app/src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt#L129-L136):

```kotlin
    private fun setSyncedFolder(syncedFolderID: Long): Boolean {
        val syncedFolderTmp = syncedFolderProvider.getSyncedFolderByID(syncedFolderID)
        if (syncedFolderTmp == null || !syncedFolderTmp.isEnabled || !syncedFolderTmp.isExisting) {
            return false
        }
        syncedFolder = syncedFolderTmp
        return true
    }
```

we can see that if `syncedFolderTmp.isExisting` is false, then false is returned.

But in [`SyncedFolder`](https://github.com/nextcloud/android/blob/master/app/src/main/java/com/owncloud/android/datamodel/SyncedFolder.java#L54), the `existing` property is documented as 

```kotlin
     * @param existing            upload existing files
```

This property does not imply that the folder should not be synced.

I can make a PR if this issue is indeed a bug, but I don't know what the original intention of checking `isExisting` was.

Please contributors, can you review this PR ? I desperately need the auto upload feature without syncing existing files.
Or maybe can you help me building a release apk locally ?

`./gradlew clean build` fails with this error :

```sh
Configuration on demand is an incubating feature.

> Configure project :app
Gradle uses Java 17.0.12-ea (Debian 17.0.12-ea+6-Debian-1)

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:compileGenericDebugJavaWithJavac'.
> Cannot query the value of this provider because it has no value available.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.9/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
```

--- a/app/src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt
+++ b/app/src/main/java/com/nextcloud/client/jobs/FilesSyncWork.kt
@@ -148,7 +148,7 @@ class FilesSyncWork(
 
     private fun setSyncedFolder(syncedFolderID: Long): Boolean {
         val syncedFolderTmp = syncedFolderProvider.getSyncedFolderByID(syncedFolderID)
-        if (syncedFolderTmp == null || !syncedFolderTmp.isEnabled || !syncedFolderTmp.isExisting) {
+        if (syncedFolderTmp == null || !syncedFolderTmp.isEnabled) {
             return false
         }
         syncedFolder = syncedFolderTmp

